pkg/backends/k6/resources.go

```diff
@@ -0,0 +1,266 @@
+package k6
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/hellofresh/kangal/pkg/backends"
+	loadTestV1 "github.com/hellofresh/kangal/pkg/kubernetes/apis/loadtest/v1"
+	"go.uber.org/zap"
+	batchV1 "k8s.io/api/batch/v1"
+	coreV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	loadTestJobName           = "loadtest-job"
+	loadTestFileConfigMapName = "loadtest-testfile"
+	loadTestDataConfigMapName = "loadtest-testdata"
+	loadTestFileVolumeName    = "loadtest-testfile-volume"
+	loadTestDataVolumeName    = "loadtest-testdata-volume"
+
+	scriptTestFileName = "test.js"
+	testdataFileName   = "testdata"
+)
+
+var (
+	loadTestLabelKey         = "app"
+	loadTestWorkerLabelValue = "loadtest-worker-pod"
+
+	defaultArgs = []string{"run", "/data/test.js"}
+)
+
+// NewJob creates a new job that runs k6
+func (b *Backend) NewJob(
+	loadTest loadTestV1.LoadTest,
+	volumes []coreV1.Volume,
+	mounts []coreV1.VolumeMount,
+	envvarSecret *coreV1.Secret,
+	reportURL string,
+	index int32,
+) *batchV1.Job {
+	logger := b.logger.With(
+		zap.String("loadtest", loadTest.GetName()),
+		zap.String("namespace", loadTest.Status.Namespace),
+	)
+
+	ownerRef := metaV1.NewControllerRef(&loadTest, loadTestV1.SchemeGroupVersion.WithKind("LoadTest"))
+
+	imageRef := fmt.Sprintf("%s:%s", loadTest.Spec.MasterConfig.Image, loadTest.Spec.MasterConfig.Tag)
+	if imageRef == ":" {
+		imageRef = fmt.Sprintf("%s:%s", b.image.Image, b.image.Tag)
+		logger.Warn("Loadtest.Spec.MasterConfig is empty; using default image", zap.String("imageRef", imageRef))
+	}
+
+	envVars := []coreV1.EnvVar{}
+	if reportURL != "" {
+		envVars = append(envVars, coreV1.EnvVar{
+			Name:  "REPORT_PRESIGNED_URL",
+			Value: reportURL,
+		})
+	}
+
+	envFrom := make([]coreV1.EnvFromSource, 0)
+	if envvarSecret != nil {
+		envFrom = append(envFrom, coreV1.EnvFromSource{
+			SecretRef: &coreV1.SecretEnvSource{
+				LocalObjectReference: coreV1.LocalObjectReference{
+					Name: newSecretName(loadTest),
+				},
+			},
+		})
+	}
+
+	args := make([]string, len(defaultArgs))
+	copy(args, defaultArgs)
+
+	if loadTest.Spec.Duration != 0 {
+		args = append(args, "--duration", loadTest.Spec.Duration.String())
```

Comment:
Could you please add a note about using `duration` request parameter to docs/k6/README.md